salt/sdb/yaml.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+'''
+Pull sdb values from a YAML file
+
+:maintainer:    SaltStack
+:maturity:      New
+:platform:      all
+
+.. versionadded:: Carbon
+
+Configuration:
+
+.. code-block:: yaml
+
+    my-yaml-file:
+      driver: yaml
+      files:
+        - /path/to/foo.yaml
+        - /path/to/bar.yaml
+
+The files are merged together and the result is searched using the same
+mechanism Salt uses for searching Grains and Pillar data structures.
+
+Optional configuration:
+
+.. code-block:: yaml
+
+    my-yaml-file:
+      driver: yaml
+      files:
+        - /path/to/foo.yaml
+        - /path/to/bar.yaml
+      merge:
+        strategy: smart
+        merge_list: false
+'''
+
+# import python libs
+from __future__ import absolute_import
+import logging
+
+import salt.exceptions
+import salt.loader
+import salt.utils
+import salt.utils.dictupdate
+
+log = logging.getLogger(__name__)
+
+__func_alias__ = {
+    'set_': 'set'
+}
+
+
+def set_(*args, **kwargs):
+    '''
+    Setting a value is not supported; edit the YAML files directly
+    '''
+    raise salt.exceptions.NotImplemented()
+
+
+def get(key, profile=None):  # pylint: disable=W0613
+    '''
+    Get a value from the REST interface
+    '''
+    data = _get_values(profile)
```

Comment:
Should we set `profile` to an empty dict here if it's not passed in? I'm just thinking that the `profile.get` call in line 76 will stacktrace with an AttributeErrpr if this is still none.